### DISCUSSION TITLE
How is a scanner paused?

### DISCUSSION THREAD

**Discussion by Clara12062** [minio/minio#19824] [Q&A] [answered]
According to the documentation at <https://min.io/docs/minio/kubernetes/upstream/operations/concepts/scanner.html>: 
> For example, by default, MinIO pauses the scanner to make I/O operations available for read and write requests. This can lengthen the time it takes for a scan to complete.

> MinIO treats the scanner as a background task and pauses it in favor of completing read and write requests on the cluster. As the cluster or workload increases, scanner performance decreases as it yields more frequently to ensure priority of normal S3 operations.

The scanner in the Minio cluster will pause when the load is high, but when looking at the source code, scanner seems to be a background processing task, and I don't see how Minio prioritizes read and write requests in high load situations. Or I could understand it this way: when the scanner's `delay` factor and `max_wait` are set to larger values, will it not cost too much system resources?

I hope someone can answer this, thank you very much!

**Comment by klauspost** [comment]
Just set `speed=slowest`.

**Reply by Clara12062** [reply]
ok, thanks!

The `speed=slowest` sets `delay`,`maxwait` and `cycle`. 
```golang
	case "slowest":
		cfg.Delay, cfg.MaxWait, cfg.Cycle = 100, 15*time.Second, 30*time.Minute
```

Could I think of scanner as something like this: scanner doesn't pause, it uses sleep intervals to make sure the scanner doesn't take up more system resources, right?

**Reply by klauspost** [reply, marked as answer]
Yes.

**Reply by Clara12062** [reply]
i get it, thanks
